crates/bitwarden-ipc/src/ipc_client.rs

```diff
@@ -38,66 +39,81 @@ where
             .await
     }
 
+    /// Receive a message
     pub async fn receive(
         &self,
     ) -> Result<IncomingMessage, ReceiveError<Crypto::ReceiveError, Com::ReceiveError>> {
         self.crypto
             .receive(&self.communication, &self.sessions)
             .await
     }
+
+    /// Receive a message, skipping any messages that cannot be deserialized into the expected
+    /// payload type.
+    pub async fn receive_typed<Payload>(
+        &self,
+    ) -> Result<TypedIncomingMessage<Payload>, ReceiveError<Crypto::ReceiveError, Com::ReceiveError>>
+    where
+        Payload: TryFrom<Vec<u8>>,
+    {
+        loop {
```

Comment:
Suggestion: It might be a nice idea to have some kind of timeout here, so we don't accidentally end up stuck on an infinite loop. Not sure how that would look like, though.

Reply:
Added timeout support :) https://github.com/bitwarden/sdk-internal/pull/187/commits/941bb068507d8c2f8b90bc43e60d45c00afcbd27